### PR TITLE
Document how to deal with Dependabot PRs

### PR DESCRIPTION
**Pull request by neilmb**
## Changes proposed in this pull request:

This PR adds some documentation on how developers should handle the dependency updates from Dependabot. @juliaklindpaintner this is my first try at documenting what I would do.

## security considerations

Just documentation changes, no security considerations

### CODE REVIEW

**Review by juliaklindpaintner — APPROVED** [review]
Thanks so much for adding this @neilmb!

File: docs/development.md [hunk 1]
```diff
@@ -244,6 +244,29 @@ redirect_from:
...
+This project uses Github's Dependabot to keep the NPM dependencies up to date.
+Dependabot takes care of noticing version updates and proposing the updates to
+`package.json` and `package-lock.json` as pull requests (PRs). A human
+developer needs to review these Dependabot PRs and merge them into the `main`
+branch.
+
+If Dependabot PRs go un-reviewed for too long, they can have merge conflicts
+and complex interactions with other code changes. The best practice is to
+review and merge Dependabot PRs as soon as possible. The automated test suite
+for this project is quite good, so this is fairly straightforward:
```
Comment (cantsin) [inline docs/development.md:258]: Do we want to add some verbiage around semvar? e.g., it is recommended to test minor or major version changes manually? (rest LGTM)

Comment (cantsin) [reply docs/development.md:258]: semvar major changes usually suggest that incompatible API changes were done, which ... could mean anything, i guess! 😓 which is why there's always an element of risk with major semvar changes. i'm not too sure how often this comes up though 🤔 

i'm fine merging as-is and if semvar issues do come up again we can re-clarify 🤷🏻‍♂️

Comment (neilmb) [reply docs/development.md:258]: I added a paragraph suggesting additional testing might be appropriate.

**Pull request merged**